File: src/js/q09.js
let boy = 21;
let girl = 11;

let ary = new Array(girl);
for (let i = 0; i < girl; i++) {
    ary[i] = new Array(boy);
    for (let j = 0; j < boy; j++) {
        ary[i][j] = 0;
    }
}
ary[0][0] = 1;
for (let i = 0; i < girl; i++) {
    for (let j = 0; j < boy; j++) {
        if ((i !== j) && ((girl - i) !== (boy - j))) {
            if (j > 0) {
                ary[i][j] += ary[i][j - 1];
            }
            if (i > 0) {
                ary[i][j] += ary[i - 1][j];
            }
        }
    }
}
console.log(ary[girl - 2][boy - 1] + ary[girl - 1][boy - 2]);

File: src/py/q09.py
def q09():
    boy = 21
    girl = 11
    ary = [0] * (boy * girl)
    ary[0] = 1

    for g in range(0, girl):
        for b in range(0, boy):
            if b != g and boy - b != girl - g:
                if b > 0:
                    ary[b + boy * g] += ary[b - 1 + boy * g]
                if g > 0:
                    ary[b + boy * g] += ary[b + boy * (g - 1)]

    print(ary[-2] + ary[-boy - 1])


if __name__ == '__main__':
    q09()
